src/auth/auth.service.ts

```diff
@@ -55,4 +66,43 @@ export class AuthService {
     return null;
   }
   
+  // verify email token
+  async confirmEmail(token: string): Promise<Boolean> {
+    let sanitizedUser = await this.usersService.findOne({emailNonce: token});
+    if (!sanitizedUser) {
+      return false;
+    }
+    return this.usersService.validateEmailNonce(token);
+  }
+
+
+  // request password reset
+  async passwordRequest(passwordRequestDto: PasswordRequestDto): Promise<void> {
+    try {
+      let user = await this.usersService.findOneUnsanitized({email: passwordRequestDto.email})
+      if (user) {
+        let passwordNonce = uuidv4()
+        await this.usersService.updatePasswordNonce(passwordNonce, user.id);
+        await this.mailService.sendPasswordReset(passwordRequestDto.email, passwordNonce);
+      }
+    } catch (err) {
+      return;
+    }
+  }
+
+  // reset password
+  async passwordReset(token: string, passwordResetDto: PasswordResetDto) : Promise<boolean> {
+    if (passwordResetDto.password !== passwordResetDto.passwordVerification) {
+      return false;
+    }
```

Comment:
Optional: could be done within the DTO using a custom decorator. This is not required for the PR to pass